packages/loader/container-loader/src/containerStorageAdapter.ts

```diff
@@ -127,12 +148,46 @@ export class ContainerStorageAdapter implements IDocumentStorageService, IDispos
 	}
 
 	public async getSnapshot(snapshotFetchOptions?: ISnapshotFetchOptions): Promise<ISnapshot> {
-		if (this._storageService.getSnapshot !== undefined) {
-			return this._storageService.getSnapshot(snapshotFetchOptions);
+		if (this._loadedGroupIdSnapshots === undefined) {
+			this._loadedGroupIdSnapshots = {};
```

Comment:
If we initialize like this, then on getting local pending state you will store these in the local pending state. But it is not necessary that we fetched snapshot with groupid as we use this method for ungrouped snapshot too. Then in if case below, "this.loadingGroupIdSnapshotsFromPendingState !== undefined" would be true which should not be the case, no?

Reply:
Good catch, fixed the problem by checking length of keys/entries instead of `undefined`.